### DISCUSSION TITLE
New version v7 <Link to="TypeSafe" />?

### DISCUSSION THREAD

**Discussion by jamalsoueidan** [remix-run/remix#10370] [Q&A]
Hi,

Thanks for the new release, looks great.

Is there any possibility to use +types (routes) to generate the urls?

I can see +types actually have the path, but I cant find how to actually use it in the frontend? Any documentation. This would ensure the frontend is not using any url that are outdated or wrong.

```js
export type Info = {
  parents: [Parent0],
  id: "routes/resume/templates"
  file: "./routes/resume/templates.tsx"
  path: "resume/:id/templates"
  params: {"id": string} & { [key: string]: string | undefined }
  module: Module
  loaderData: T.CreateLoaderData<Module>
  actionData: T.CreateActionData<Module>
}

export namespace Route {
  ...
}
```

**Comment by sergiodxa** [comment]
The `<Link to>` prop is not type-safe yet but the team has said they're working on it.

**Reply by jamalsoueidan** [reply]
They really take their time 😄 Thanks for your reply.
